### ISSUE TITLE
Phase 1 Testing

### ISSUE THREAD

**Issue by riongull** [dashhive/wallet-ui#21]
### Phase 1 Testing

> Testing dates:  Feb 1, 2024 - Feb 5, 2024

This was originally authored as a [Google Doc](https://docs.google.com/document/d/1eZAK9tRvxe5l5DGOC95-fp_Rc7YoKMUlnC-HIU-1YZg/edit), converting to a GitHub issue because @jojobyte likes Microsoft more than Google (</sarcasm>)

Most of these issues won’t need dedicated GitHub issues or PRs, but I will categorize them like [conventional commits](https://www.conventionalcommits.org/) to help organize them.  Some user experience (UX) issues will be tagged as “feat” if they affect a current feature that may need to be reconsidered, others may be more simple changes, which could go under “style” or “doc”.

I’m basically going through the whole app and testing each feature (link, button, etc).  I’m not necessarily trying to break the app, but I’m monkeying around and being more picky than a normal user.

This document incorporates feedback from me and [juliushamilton1](https://www.youtube.com/watch?v=l9DrzuoyCjc)

#### Welcome screen

- [ ] feat: Can’t tab through buttons.
- [ ] I want to be able to navigate to either button and hit enter and have it work.
- [ ] style: Some text is redundant, but no need to change for now.
- [ ] fix:  I tried “Generate a New Wallet” first, then exited out after some testing, then tried to click “Add an Existing Wallet” and I got no feedback.  Had to refresh to get it to work again.
- [ ] style(julius): Is there some reason it is necessary to use the terms “disconnect” and “add wallet” as opposed to just “log in” and “log out”?

#### New Wallet modal

- [ ] doc: There is no description about what the Alias is and what I should do here.  A description does exist in the “Existing Wallet” modal (also mentioned below).
- [ ] doc: The error field could be more helpful by describing (in plain English) what the requirements are for a valid alias (e.g. “numbers and letters only, at least 3 characters”)
- [ ] doc(julius): Minor aspects of user understanding: how does the wallet actually work? What information is your browser storing when you create a wallet? Is it downloading a temporary self custodial wallet into the browser session? How does that work, where is it saved to? How does locking work?
- [ ] doc(julius): User experience: should you be able to create a new alias every time you lock back in?

#### Existing Wallet modal

- [ ] style: Underlined “Keystore” and “Backup” makes me think they are separate links that do different things.  I know from talking with jojobyte that it’s just one button that does the same thing, but let’s try to not trick people.  Maybe just make them bold (remove the underline, and remove the bold from the “or”).
- [ ] fix:  The seed phrase eye toggle doesn’t seem to work (at all, or at least reliably). 
- [ ] fix(julius): the show/hide button for the seed wallet entry field doesn’t work
- [ ] style: After I’ve picked a file (or dropped one into the field) I want the “Seed Phrase” field to go away since it will never be needed in that case, and the user shouldn’t feel like he’s leaving something blank.  If I’ve picked a “backup” then the alias should go away also, since it is always required when making a new wallet (is it enforced when backing up?). If so, it would be populated after hitting “Add wallet” and the user can always edit it, but I don’t think we should help them get in the habit of redefining the alias here in the “Existing Wallet” import.  An alternative to hiding the alias field would be to name it automatically by the file name, but that might require more coding than it’s worth, and it might clash with the name in the keystore file, so I’m not sure that’s a good option. 
- [ ] doc: Here’s a good description of the alias.  Let’s copy this over to the other modal.

#### Decrypt Wallet modal

- [ ] feat: I’d like the “Remember Password for Browser Session” to be toggled on by default. I know it’s potentially less secure, but at least for now while we’re in alpha (with our warning already telling them they’re using at their own risk) I’d like to fall on the side of favoring UX.
- [ ] fix: This isn’t a mobile review, but I seem to have noticed that I always have to unlock my wallet on mobile.  Maybe there’s a bug on mobile?

#### Main screen

- [ ] style(julius): Minor visual design notes: the landing page feels slightly “empty”. Awkward side columns.
- [ ] feat: Same as welcome screen item 1.
- [ ] I want all buttons, fields, links, etc accessible by keyboard navigation
- [ ] I can see it going from #home to #me, and hitting enter works, but not all action options (like the buttons) are available.
- [ ] feat: I kind of want some hover options right on each contact in the list.  This would bring some life and color to the otherwise sparse list.
- [ ] send arrow - shortcut to send to this contact
- [ ] receive arrow - shortcut to request from this contact
- [ ] trash icon - quickly get rid of contacts that I know I didn’t share
- [ ] star - not now, but later on, to bring up to a top section of the list
- [ ] feat: Maybe we should use the big empty space in the contacts list to recommend pairing with other prominent wallets (mobile wallet, Desktop wallet, etc).  
- [ ] DCG’s mobile wallet definitely needs to display a QR for its xpub address.
- [ ] We need to support importing raw xpubs if we don’t already. 

#### Alias Button

- [ ] style: The alias button at the top right isn’t obviously clickable/actionable until you’re already hovering it, and even then you only get cursor feedback. 
- [ ] Edit Profile modal
- [ ] feat: I don’t think we should have the “Use this QR code to fund your wallet” section here.  Might be better in “Request” (which maybe needs to be renamed “Receive”

#### Backup Wallet modal

- [ ] style: Top button is “adjective noun”, bottom two buttons are “verb noun”.  Maybe make it consistent.

Lock modal

- [ ] style: The bottom button lines don’t line up, maybe because of emoji in one but not the other.  Or maybe because of box-sizing: border-box

#### Disconnect modal

- [ ] style: Still not sure “disconnect” is the right word.  Leave for now.  Either way, maybe change the bomb to something else?  A broom or something else could be used if we changed it to “Clean”, but I’m not sure there’s anything better than a bomb for now.  Leave it for now.  

#### Dash Balance field

- [ ] feat: Add a refresh button, maybe.  It shouldn’t really be needed, right, but sometimes it is, and users will probably want it.

#### Send Funds modal

- [ ] feat: No way to send 100%.  Do our SDKs facilitate this?
- [ ] fix: Why does Scanning only do “Scan a Contact”
- [ ] Scanning a normal Dash address captures it, but doesn’t parse and enter in the to field.
- [ ] I don’t like the up/down buttons on the input field
- [ ] Does anyone adjust the amount like that?
- [ ] Either way, we should make the step = 0.001 DASH (last digit in white area, about 3 cents USD)
- [ ] 

#### Request Funds modal

- [ ] feat: Still not sure where the best place is for a QR code to fund my own wallet.  Usually it’s right there available when a user hits “Receive” in his wallet.  I think that’s probably the most intuitive place due to people being use to it in other wallets.  Currently we have that feature under MyAlias > Edit Profile.  I don’t think that’s the right place for it.
- [ ] feat: Every time I click Request the wallet generates a new address.  This might be unsettling for users.  Need to rethink this functionality.  Side note/question: Is this cycling through the 0 index on the HD path when no contact is selected?  Either way, we may want to put our own username in the alias input box by default, which would be the default funding mechanism.
- [ ] 

#### Add a New Contact modal

- [ ] Overall, this modal seems too busy and is not intuitive enough.  There are two main (and very distinct) expectations of what to do to “add a contact”:  1) Initiate a contact request, and 2) follow up on a contact request.  When I click this it’s not obvious what I’m supposed to (or at least I don’t think I would if I were new using this wallet).  It’s not terrible, and I can figure it out (even as a newb), but it’s not obvious.
- [ ] Send / Share your contact info
- [ ] Send Link (copy/paste)
- [ ] Display QR (enlarge image)
- [ ] Receive / Accept / Import someone’s contact info
- [ ] Receive Link (copy/paste)
- [ ] Scan QR (camera button)
- [ ] What about putting data into a normal web link ([https://wallet.dashincubator.dev](https://wallet.dashincubator.dev/#home) with parameters in the query string) that people can just click and accept your friend request)?
- [ ] The QR reader doesn’t accept/scan my xpub address from my iPhone mobile wallet.  
- [ ] I made a QR code out of the text, but it didn’t read it.
- [ ] It does show a green square when you line it up right, but it doesn’t do anything other than that.
- [ ] Adding contacts doesn’t seem to be working at all.  Saving has two issues:
- [ ] The address/xpub isn’t saving
- [ ] The name is appending “- Finish pairing with contact” to every alias, even if adding all info in one fresh and complete step.
- [ ] Might only be when using a raw xpub (text starting with “xpub…”

#### Clicked Contact modal

- [ ] style: The modal could use a header like I think all the other modals have.

**Comment by coolaj86** [comment]
- Needs a way to skip the passphrase (i.e. have an empty string `""` or a constant, such as `"NONE"`)
- We need a way to see the list of what money has been sent and received
- I shouldn't have to click Send 3 times to send \
  (but there definitely should be an animation and a 2 second "sending" - even if it doesn't take that long)
